kernel/src/process.rs

```diff
@@ -375,7 +375,7 @@ impl TbfHeader {
         match *self {
             TbfHeader::TbfHeaderV1(hd) => hd.entry_offset,
             TbfHeader::TbfHeaderV2(hd) =>
-                hd.main.map_or(0, |m| m.init_fn_offset) + (hd.base.header_size as u32),
+                hd.main.map_or(0, |m| m.init_fn_offset) + (hd.base.header_size as u32) + self.get_total_writeable_flash_size(),
```

Comment:
I think per the documentation of the TBF header, we don't want to change this. There is no requirement that apps put their writeable flash before the text or in one contiguous block. `init_fn_offset` should just be set correctly when the header is generated (e.g. by elf2tbf).